runtime/vm.go

```diff
@@ -674,9 +698,32 @@ func (v *VM) run() {
 				v.sp = v.sp - numArgs + callee.Fn.NumLocals
 
 			case *objects.CompiledFunction:
+				// consolidate args into an array
+				if callee.VarArgs {
+					realArgs := callee.NumParameters - 1
```

Comment:
Perhaps rewrite as the following to avoid unnecessary computations at runtime (and similarly for Closures):

```
realArgs := callee.NumParameters - 1
varArgs := numArgs - realArgs
if varArgs < 0 {
	goto wrongNumberOfParametersCompiledFunction
}
sp := v.sp - varArgs
numArgs = realArgs + 1
args := make([]objects.Object, 0, varArgs)

for i := sp; i < v.sp; i++ {
	args = append(args, v.stack[i])
}

v.stack[sp] = &objects.Array{Value: args}
v.sp = sp + 1
```